File: 002/fibonacci.go
package main

import "fmt"

func main() {
	sum := 0
	for a, b := 1, 2; b <= 4e6; a, b = b, a+b {
		if b%2 == 0 {
			sum += b
		}
	}
	fmt.Println(sum)
}

File: 002/fibonacci.rs
fn main() {
    let mut sum = 0;
    let mut a = 1;
    let mut b = 2;
    let mut tmp;
    while b <= 4000000 {
        if b % 2 == 0 {
            sum += b;
        }
        tmp = a + b;
        a = b;
        b = tmp;
    }
    println!("{}", sum);
}
